commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/input/CharCodeWithModifiers.java

```diff
@@ -64,4 +64,24 @@ public static int computeKeyDigest(int modifiers, int charCode) {
     public static int computeKeyDigest(SignalEvent event) {
         return computeKeyDigest(ModifierKeys.computeModifiers(event), KeyCodeMap.getKeyFromEvent(event));
     }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        CharCodeWithModifiers that = (CharCodeWithModifiers)o;
+
+        if (modifiers != that.modifiers) return false;
+        return charCode == that.charCode;
```

Comment:
I think it will be better to write 
return modifiers == that.modifiers
          && charCode == that.charCode;